### ISSUE TITLE
[HoverCard] keyboard support

### ISSUE THREAD

**Issue by joshwcomeau** [radix-ui/primitives#898]
### Overview

As I understand it, HoverCard is intended to provide a visual preview for sighted users, and for that reason, it's mouse-only, with no keyboard controls. I'd love to see it extended to cover focus as well, either by default or through a boolean `triggerOnFocus` prop.

### Who does this impact? Who is this for?

This would help sighted users who don't use a pointer device. I was in this category myself for a while in 2020, using voice dictation to navigate through the web. It'd help anyone who uses non-pointer input devices, like keyboards or switches.

### Additional context

It may be that this component really isn't meant for keyboard users / there's another component that works better for them (maybe `Popover`?). In this case, I think maybe the docs could help clarify this, and make clear exactly when each component should/shouldn't be used.

The way it's currently framed, I worry developers will lump two very-different groups of people together: unsighted folks who use a screen-reader, and sighted folks who use a non-pointer input device. And I think it's worth making sure that we optimize the experience for both of them!

Overall I'm a big fan of this library 💯 Thanks for all the great work!

**Comment by benoitgrelard** [comment]
Hey Josh, thanks for creating this!

`HoverCard` is by design not for screen reader users, because its interaction model doesn't allow for proper focus management and enunciation of the content inside. Also it should only be used to show redundant information that is already available behind the link (trigger).

That said I think you are making a really good point about the fact that right now it is quite possibly too binary (sighted users vs. screen reader users). There's definitely an argument to be made for a sighted person that cannot use the pointer so cannot hover.

However, given that the information behind it should be redundant, and not primary info that isn't available any other way, there's also an argument for not having something pop-up everywhere that you move your focus to. I imagine for example if I browsed twitter or github with my keyboard tabbing through, and the "user" card popped up everytime I landed on a name, etc that might also be annoying.

In that sense, an approach where the devs could decide if they want to enable that or not might be good.

> In this case, I think maybe the docs could help clarify this, and make clear exactly when each component should/shouldn't be used.

This is something we have been talking about a lot, and we need to get better at in our docs for sure.

**Comment by jjenzz** [comment]
> But since these elements aren't meant to contain focusable elements

As far as I understand it, `HoverCard` can contain focusable links, however, if we kept the card out of the tab sequence (portalled) and just rendered it on link focus, would that cover your expectation?

I initially thought the request was to make the card keyboard accessible too which I think we are all agreeing would be frustrating. Having the card appear out of sequence on focus is definitely doable and seems sensible to me.
